core/src/test/java/io/grpc/ConnectivityStateInfoTest.java

```diff
@@ -88,6 +88,7 @@ public void equality() {
     assertNotEquals(info4, info6);
 
     assertFalse(info1.equals(null));
-    assertFalse(info1.equals(this));
+    // Extra cast to avoid ErrorProne EqualsIncompatibleType failure
+    assertFalse(((Object) info1).equals(this));
```

Comment:
Do you know which `equals()` method `assertNotEquals()` uses? It looks unspecified to me, so calling `equals()` manually would be the only safe way to do the test.